### DISCUSSION TITLE
How can I configure my project to only build the "staging" branch?

### DISCUSSION THREAD

**Discussion by thinkswan** [vercel/vercel#5787] [Help]
Hi Vercel community! 👋 

I have a project hosted on the platform, but I'd only like builds to be triggered for a specific branch. I read about the ["Ignored Build Step" option](https://vercel.com/docs/platform/projects#ignored-build-step), as well as the relevant [KB article](https://vercel.com/knowledge/how-do-i-use-the-ignored-build-step-field-on-vercel) and [previous discussions](https://github.com/vercel/vercel/discussions/5171#discussioncomment-137479), but I can't quite figure it out.

**My mission:** Have my project only trigger builds for the `staging` branch.

**Attempt 1:** Set the ignored build step to `echo $VERCEL_GIT_COMMIT_REF | grep --quiet staging && exit 1 || exit 0`. This works in my shell, but when Vercel runs it, the build is cancelled with the following output:

```
13:21:36.502 | Cloning github.com/REDACTED/webapp (Branch: staging, Commit: cdfdd19)
13:21:38.447 | Running "echo $VERCEL_GIT_COMMIT_REF | grep --quiet staging && exit 1 || exit 0"
13:21:38.453 | $VERCEL_GIT_COMMIT_REF | grep --quiet staging && exit 1 || exit 0
13:21:38.453 | The Deployment has been canceled as a result of running the command defined in the "Ignored Build Step" setting.
13:21:39.273 | Deployment has been canceled
```

**Attempt 2:** Set the ignored build step to `bash -c 'if [[ "$VERCEL_GIT_COMMIT_REF" == "staging" ]] ; then exit 1; else exit 0; fi`. Again, this works in my shell, but when Vercel runs it, the result is a syntax error.

```
13:43:49.080 | Cloning github.com/REDACTED/webapp (Branch: task/test-ci, Commit: 7cdb126)
13:43:49.996 | Running "bash -c 'if [[ "$VERCEL_GIT_COMMIT_REF" == "staging" ]] ; then exit 1; else exit 0; fi'"
13:43:50.002 | [[: -c: line 0: unexpected EOF while looking for matching `''
13:43:50.002 | [[: -c: line 1: syntax error: unexpected end of file
```

I made sure the "Automatically expose System Environment Variables" option was turned on and I also tried adding a system environment variable that pointed to the `VERCEL_GIT_COMMIT_REF` variable. None of these worked.

How can I configure my project to only build the `staging` branch?

**Comment by mledwards** [comment]
I adapted [their docs](https://vercel.com/guides/how-do-i-use-the-ignored-build-step-field-on-vercel) to check for `VERCEL_GIT_COMMIT_REF` instead of `VERCEL_ENV`. Yours would look something like this:

```
if [[ "$VERCEL_GIT_COMMIT_REF" == "staging" ; then
  # Proceed with the build
    echo "✅ - Build can proceed"
  exit 1;

else
  # Don't build
  echo "🛑 - Build cancelled"
  exit 0;
fi
```
